### PR TITLE
CB-3349 Blueprint list needs a queryparameter to list sdx blueprints

### PR DESCRIPTION
**Pull request by topolyai5**

### CODE REVIEW

File: core/src/main/java/com/sequenceiq/cloudbreak/service/blueprint/BlueprintService.java [hunk 1]
```diff
@@ -176,11 +176,23 @@ public PlatformRecommendation getRecommendation(Long workspaceId, String bluepri
...
         updateDefaultBlueprintCollection(workspace);
-        return blueprintViewRepository.findAllByNotDeletedInWorkspace(workspaceId);
+        Set<BlueprintView> allByNotDeletedInWorkspace = blueprintViewRepository.findAllByNotDeletedInWorkspace(workspaceId);
+        if (withSdx) {
+            return allByNotDeletedInWorkspace;
+        }
+        return allByNotDeletedInWorkspace.stream().filter(it -> !isSDXReady(it)).collect(Collectors.toSet());
+    }
+
+    private boolean isSDXReady(BlueprintView blueprintView) {
```
Comment (gergopapi2) [inline core/src/main/java/com/sequenceiq/cloudbreak/service/blueprint/BlueprintService.java]: Abbreviation SDX I would write as Sdx in a method name.

File: core/src/main/java/com/sequenceiq/cloudbreak/service/blueprint/BlueprintService.java [hunk 2]
```diff
@@ -176,11 +176,23 @@ public PlatformRecommendation getRecommendation(Long workspaceId, String bluepri
...
+        if (blueprintView.getTags() == null || blueprintView.getTags().getValue() == null) {
+            return false;
+        }
+        Boolean sdxReady = blueprintView.getTags().getValue("shared_services_ready" );
```
Comment (gergopapi2) [inline core/src/main/java/com/sequenceiq/cloudbreak/service/blueprint/BlueprintService.java]: There is already a string constant for "shared_services_ready", I would use that.

File: core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/blueprint/BlueprintV4Endpoint.java [hunk 3]
```diff
@@ -38,7 +40,7 @@
     @Produces(MediaType.APPLICATION_JSON)
     @ApiOperation(value = BlueprintOpDescription.LIST_BY_WORKSPACE, produces = JSON, notes = BLUEPRINT_NOTES,
             nickname = "listBlueprintsByWorkspace")
-    BlueprintV4ViewResponses list(@PathParam("workspaceId") Long workspaceId);
+    BlueprintV4ViewResponses list(@PathParam("workspaceId") Long workspaceId, @DefaultValue("false") @QueryParam("withSdx") Boolean withSdx);
```
Comment (doktoric) [inline core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/blueprint/BlueprintV4Endpoint.java]: withDatalake

Comment (topolyai5) [reply core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/blueprint/BlueprintV4Endpoint.java]: we use everywhere sdx (on public-facing), why should we use the datalake here?

**Pull request merged**